File: testdata/bench/rust/BenchmarkTest00083.rs
use actix_web::{web, HttpResponse};
use std::collections::HashMap;

// Vulnerable: XSS via div injection
async fn handler(query: web::Query<HashMap<String, String>>) -> HttpResponse {
    let param = query.get("msg").unwrap_or(&String::new()).clone();
    let html = format!("<div class='result'>{}</div>", param);
    HttpResponse::Ok().content_type("text/html").body(html)
}

File: testdata/bench/cpp/BenchmarkTest00083.cpp
#include <cstdio>
#include <cstdlib>

int main() {
    char *input = getenv("QUERY_STRING");
    char response[4096];
    sprintf(response, "<html><body><div>%s</div></body></html>", input);
    printf("Content-Type: text/html\r\n\r\n%s", response);
    return 0;
}
